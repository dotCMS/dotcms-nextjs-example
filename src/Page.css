.row {
    border: solid 1px;

}

.column {
    border: solid 2px;
    min-height: 100px;
}