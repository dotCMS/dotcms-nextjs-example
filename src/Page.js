import React from 'react';

import RowContainer from './Components/RowContainer';
import CantRender from './Components/CantRender';

import './Page.css';



const Page = ({ data }) => {
    return data && data.body && data.body.rows ? (
        data.body.rows.map((row, i) => {
            return <RowContainer row={row} key={i} />;
        })
    ) : (
        <CantRender title="Page can not be rendered">
            <p>You might not have enterprise license or the page is not created with our layout editor.</p>
        </CantRender>
    );
};

export default Page;
