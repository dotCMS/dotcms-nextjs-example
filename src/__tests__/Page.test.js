import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Page from '../Page';

describe('<Page />', () => {
    it('renders rows correctly', () => {
        const wrapper = shallow((<Page data={{body: {rows: [{}, {}]}}} />));
        const rows = wrapper.find('RowContainer');
        expect(rows.length).toBe(2);
    });

    it('should render alert', () => {
        const wrapper = shallow((<Page data={{body: {}}} />));
        const cantRender = wrapper.find('CantRender');
        expect(cantRender.length).toBe(1);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

})
