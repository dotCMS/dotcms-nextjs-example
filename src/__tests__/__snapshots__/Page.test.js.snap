// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render alert 1`] = `
<CantRender
  title="Page can not be rendered"
>
  <p>
    You might not have enterprise license or the page is not created with our layout editor.
  </p>
</CantRender>
`;
